spring-data-geode-test/src/main/java/org/springframework/data/gemfire/tests/integration/ForkingClientServerIntegrationTestsSupport.java

```diff
@@ -53,70 +51,75 @@
 @SuppressWarnings("unused")
 public abstract class ForkingClientServerIntegrationTestsSupport extends ClientServerIntegrationTestsSupport {
 
-	private static ProcessWrapper gemfireServer;
+    private static List<ProcessWrapper> processes = new LinkedList<>();
+
+    public static void startGemFireServer(Class<?> gemfireServerConfigurationClass, String... arguments)
+            throws IOException {
 
-	public static void startGemFireServer(Class<?> gemfireServerConfigurationClass, String... arguments)
-			throws IOException {
+        int availablePort = setAndGetPoolPortProperty(setAndGetCacheServerPortProperty(findAvailablePort()));
 
-		int availablePort = setAndGetPoolPortProperty(setAndGetCacheServerPortProperty(findAvailablePort()));
+        List<String> argumentList = new ArrayList<>(Arrays.asList(nullSafeArray(arguments, String.class)));
 
-		List<String> argumentList = new ArrayList<>(Arrays.asList(nullSafeArray(arguments, String.class)));
+        argumentList.add(String.format("-D%s=%d", GEMFIRE_CACHE_SERVER_PORT_PROPERTY, availablePort));
 
-		argumentList.add(String.format("-D%s=%d", GEMFIRE_CACHE_SERVER_PORT_PROPERTY, availablePort));
+        addGemFireServerProcess(run(gemfireServerConfigurationClass, argumentList.toArray(new String[0])));
 
-		setGemFireServerProcess(run(gemfireServerConfigurationClass, argumentList.toArray(new String[0])));
+        waitForServerToStart("localhost", availablePort);
+    }
 
-		waitForServerToStart("localhost", availablePort);
-	}
+    protected static int setAndGetCacheServerPortProperty(int port) {
 
-	protected static int setAndGetCacheServerPortProperty(int port) {
+        System.setProperty(GEMFIRE_CACHE_SERVER_PORT_PROPERTY, String.valueOf(port));
 
-		System.setProperty(GEMFIRE_CACHE_SERVER_PORT_PROPERTY, String.valueOf(port));
+        return port;
+    }
 
-		return port;
-	}
+    protected static int setAndGetPoolPortProperty(int port) {
 
-	protected static int setAndGetPoolPortProperty(int port) {
+        System.setProperty(GEMFIRE_POOL_SERVERS_PROPERTY, String.format(GEMFIRE_LOCALHOST_PORT, port));
 
-		System.setProperty(GEMFIRE_POOL_SERVERS_PROPERTY, String.format(GEMFIRE_LOCALHOST_PORT, port));
+        return port;
+    }
 
-		return port;
-	}
+    @AfterClass
+    public static void stopGemFireServer() {
```

Comment:
This needs to be synchronized; it is a compound action involving the List of GemFire Server ProcessWrappers.

Reply:
addressed